SRA_Sensors/SRA_Sensors.ino

```diff
@@ -65,29 +94,66 @@ void updateLed(int msg)
     Computer_Serial.println("Setting Low");
   }
 }
-void o2Reading(){
-  for(int i=0;i<25;i++)
-  {
-    if(O2_Serial.read()=='O') //Start at the beginning of the o2 sensor output
-    {
-      float o2readings[4];
-      readO2Bytes(1);
-      o2readings[0] =  strtof(readO2Bytes(6).c_str(),NULL); //Partial Pressure in mBar?
-      readO2Bytes(4); 
-      o2readings[1] = strtof(readO2Bytes(4).c_str(),NULL); //Temperature in celsius
-      readO2Bytes(3);
-      o2readings[2] = strtof(readO2Bytes(4).c_str(),NULL); //Pressure in mBar?
-      readO2Bytes(3);
-      o2readings[3] = strtof(readO2Bytes(6).c_str(),NULL)*10000; //Concentration - read in percent, converted to ppm
-      readO2Bytes(11);
 
+void pdReading()
+{
+  float pdreadings[3];
+
+  //gets adc values from sensor output
+  uint16_t raw1 = analogRead(Photodiode1);
+  uint16_t raw2 = analogRead(Photodiode2);
+  uint16_t raw3 = analogRead(Photodiode3);
+
+  // get voltage values
+  float v1 = raw1 * ref_voltage / adc_resolution;
+  float v2 = raw2 * ref_voltage / adc_resolution;
+  float v3 = raw3 * ref_voltage / adc_resolution;
+
+  Serial.println((float)v1);
+  Serial.println((float)v2);
+  Serial.println((float)v3);
+
+  //get current values from voltage values from formula
+  pdreadings[0] = reverseCurrent*(exp(-(50/3) * v1)-1) - darkCurrent; 
+  pdreadings[1] = reverseCurrent*(exp(-(50/3) * v2)-1) - darkCurrent;
+  pdreadings[2] = reverseCurrent*(exp(-(50/3) * v3)-1) - darkCurrent;
+
+  RoveComm.write(RC_SCIENCESENSORSBOARD_FLUOROMETERDATA_DATA_ID,RC_SCIENCESENSORSBOARD_FLUOROMETERDATA_DATA_COUNT, pdreadings);
+}
+
+void o2Reading()
+{
+    if(O2_Serial.read()=='%') //Start at the beginning of the o2 Percentage sensor output
+    {
+      float o2readings[1];
+      O2_Serial.read();
+      o2readings[0] = strtof(readO2Bytes(6).c_str(),NULL)*10000.0; //Concentration - read in percent, converted to ppm
       RoveComm.write(RC_SCIENCESENSORSBOARD_O2_DATA_ID,RC_SCIENCESENSORSBOARD_O2_DATA_COUNT, o2readings);
-      break;
     }
-  }
 }
 
-void co2Reading(){
+void ch4Reading()
+{
+    if(userial.read() =='C') //Start at the beginning of the ch4 sensor output
+    {
+      if(userial.read() ==':') //Start at the beginning of the ch4 sensor output
+      {
+        userial.read();
```

Comment:
(Repeat this for all lines like this one) Maybe add a comment to explain why you're throwing away a byte. I know it's to skip a space, but should be clear about it.